sips/fork_support.md

```diff
@@ -20,56 +20,254 @@ Client implementation should calculate when a new duty starts to decide if it st
+		return mainnetForks()
+	case JatoNetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoTestnet}}
+	case JatoV2NetworkID:
+		return []*ForkData{{Epoch: 0, Domain: JatoV2Testnet}}
+	default:
+		return []*ForkData{}
+	}
 }
 
-func genesisForks() []*ForkData {
-    return []*ForkData{
-        {
-            Epoch:  0,
-            Domain: GenesisMainnet,
-        },
-    }
+// mainnetForks returns all forks for the mainnet chain
+func mainnetForks() []*ForkData {
+	return []*ForkData{
+		{
+			Epoch:  0,
+			Domain: GenesisMainnet,
+		},
+	}
+}
+
+func (networkID NetworkID) DefaultFork() *ForkData {
+	return networkID.GetForksData()[0]
+}
+
+// GetCurrentFork returns the ForkData with highest Epoch smaller or equal to "epoch"
+func (networkID NetworkID) ForkAtEpoch(epoch phase0.Epoch) (*ForkData, error) {
+	// Get list of forks
+	forks := networkID.GetForksData()
+
+	// If empty, raise error
+	if len(forks) == 0 {
+		return nil, errors.New("Fork list by GetForksData is empty. Unknown Network")
+	}
+
+	var current_fork *ForkData
+	for _, fork := range forks {
+		if fork.Epoch <= epoch {
+			current_fork = fork
+		}
+	}
+	return current_fork, nil
+}
+
+func (f ForkData) GetRoot() ([]byte, error) {
+	byts, err := json.Marshal(f)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not marshal ForkData")
+	}
+	ret := sha256.Sum256(byts)
+	return ret[:], nil
+}
+```
+
+### Structures access to types
+
+- `Share`: It's more appropriate for the structure to hold the information on the network that it's in, instead of the domain type.
+    Thus, _Share_ will contain a `NetworkID` field instead of a `DomainType` field.
+
+Regarding the indirect usage of Identifiers:
+- `Instance`:
+  - Must keep its identifier to send properly formed messages.
+- `Controller`:
+  - Should replace its _Identifier_ with an _IdentifierF_ getter function, passed by the Runner. With the function instead of the fixed value, it can compute the up-to-date identifier when needed.
+  - Since the controller's `config` must also be updated, the controller should also contain a _DomainTypeF_ function, instead of its fixed _DomainType_ field.
+
+```go
+
+type IdentifierF func() []byte // <-- new func type
+type DomainTypeF func() (types.DomainType, error) // <-- new func type
+
+type Controller struct {
+	IdentifierF IdentifierF // <-- replace Identifier []byte to Identifier getter function
+	Height     Height 
+	StoredInstances InstanceContainer
+	FutureMsgsContainer map[types.OperatorID]Height
+	DomainTypeF         DomainTypeF // <-- replace Domain types.DomainType to DomainType getter function
+	Share               *types.Share
+	config              IConfig
+}
+
+func (c *Controller) StartNewInstance(height Height, value []byte) error {
+	// ...
+
+	domainType, err := c.DomainTypeF()
+	if err != nil {
+		return Errors.Wrap(err,"Coudld not get current DomainType.")
+	}
+	c.config.SetSignatureDomainType(domainType) // <-- updates its config using its new DomainF
```

Comment:
config is being injected into the controller and provides data/ functions. The controller shouldn't be "setting" stuff in the config